compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/AnnotationDesugar.java

```diff
@@ -166,6 +171,9 @@ private void defineClassAnnotations(BLangPackage pkgNode, SymbolEnv env, BLangFu
             }
 
             BLangClassDefinition classDefinition = (BLangClassDefinition) topLevelNode;
+            if (isServiceDeclaration(classDefinition)) {
+                addIntrospectionInfoAnnotation(classDefinition);
```

Comment:
@rdhananjaya Usually when annotations are added to a service decl, it seems it gets added to both the `BLangService` node and the corresponding `BLangClassDefinition` node. But here it seems the annotation is added only to the generated class. Is this ok? 🤔

Reply:
This should be OK as they only need this at runtime. I think with this we will miss this annotation in BIR.